src/aosm/azext_aosm/generate_nsd/nsd_generator.py

```diff
@@ -0,0 +1,212 @@
+        self.nsd_bicep_output_name = NSD_DEFINITION_BICEP_FILE
+
+        self.build_folder_name = self.config.build_output_folder_name
+
+    def generate_nsd(self, deploy_parameters) -> None:
+        """Generate a NSD templates which includes an Artifact Manifest, NFDV and NF templates."""
+        logger.info(f"Generate NSD bicep template")
+
+        self.deploy_parameters = deploy_parameters
+
+        self._create_nsd_folder()
+        self.create_parameter_files()
+        self.write_nsd_manifest()
+        self.write_nf_bicep()
+        self.write_nsd_bicep()
+
+        print(f"Generated NSD bicep templates created in {self.build_folder_name}")
+        print(
+            "Please review these templates. When you are happy with them run "
+            "`az aosm nsd publish` with the same arguments."
+        )
+
+    def _create_nsd_folder(self) -> None:
+        """
+        Create the folder for the NSD bicep files.
+
+        :raises RuntimeError: If the user aborts.
+        """
+        if os.path.exists(self.build_folder_name):
+            carry_on = input(
+                f"The folder {self.build_folder_name} already exists - delete it and continue? (y/n)"
+            )
+            if carry_on != "y":
+                raise RuntimeError("User aborted!")
+
+            shutil.rmtree(self.build_folder_name)
+
+        logger.info("Create NFD bicep %s", self.build_folder_name)
+        os.mkdir(self.build_folder_name)
+
+    def create_parameter_files(self) -> None:
+        """Create the Schema and configMappings json files."""
+        schemas_folder_path = os.path.join(self.build_folder_name, SCHEMAS)
+        os.mkdir(schemas_folder_path)
+        self.write_schema(schemas_folder_path)
+
+        mappings_folder_path = os.path.join(self.build_folder_name, CONFIG_MAPPINGS)
+        os.mkdir(mappings_folder_path)
+        self.write_config_mappings(mappings_folder_path)
+
+    def write_schema(self, folder_path: str) -> None:
+        """
+        Write out the NSD Config Group Schema JSON file.
+
+        :param folder_path: The folder to put this file in.
+        """
+        logger.debug(f"Create {self.config.cg_schema_name}.json")
+
+        schema_path = os.path.join(folder_path, f"{self.config.cg_schema_name}.json")
+
+        with open(schema_path, "w") as _file:
+            _file.write(self.deploy_parameters)
+
+        logger.debug(f"{schema_path} created")
+
+    def write_config_mappings(self, folder_path: str) -> None:
+        """
+        Write out the NSD configMappings.json file.
+
+        :param folder_path: The folder to put this file in.
+        """
+
+        deploy_parameters_dict = json.loads(self.deploy_parameters)
+        deploy_properties = deploy_parameters_dict["properties"]
+
+        logger.debug("Create configMappings.json")
+        config_mappings = {
+            key: f"{{{self.config.cg_schema_name}.{key}}}" for key in deploy_properties
```

Comment:
I think this line is wrong. You need to have {configurationparameters.('{self.config.cg_schema_name}').{key}} or something similar